app/scripts/services/paymentReportService.js

```diff
@@ -0,0 +1,95 @@
+angular
+  .module('confRegistrationWebApp')
+  .service('paymentReportService', function PaymentReportService(
+    $cacheFactory,
+    $rootScope,
+    $http,
+    $q,
+  ) {
+    const path = function(id) {
+      return 'conferences/' + (id || '') + '/payments/report';
+    };
+
+    function getReportData(url) {
+      const defer = $q.defer();
+      $rootScope.loadingMsg = 'Loading Details';
+      $http
+        .get(url)
+        .then(function(response) {
+          var report = response.data;
+          defer.resolve(report);
+        })
+        .catch(function(error) {
+          defer.reject(error);
+        })
+        .finally(function() {
+          $rootScope.loadingMsg = '';
+        });
+      return defer.promise;
+    }
+
+    this.getAll = function(id) {
+      var allReportsUrl = path(id);
+      return getReportData(allReportsUrl);
+    };
+
+    this.getReport = function(conferenceId, queryParameters) {
+      var defer = $q.defer();
```

Comment:
Return the $http promise instead of using defer.

Reply:
done